src/nemo_evaluator/solvers/harbor.py

```
@@ -1203,6 +1260,16 @@ async def solve(
                     raise InfraError(f"Agent infrastructure failure: {etype}: {agent_error}") from agent_error
                 error = f"Agent crashed: {etype}: {agent_error}"
                 logger.warning("HarborSolver: %s", error)
+            elif (_crash_file := agent_logs_dir / "nel_agent_error.json").is_file():
+                try:
+                    _crash = json.loads(_crash_file.read_text())
+                    _etype = _crash.get("etype", "AgentCrash")
+                    _emsg = _crash.get("emsg", "")
+                    if _etype not in _infra_error_names:
+                        error = f"Agent crashed: {_etype}: {_emsg}"
+                        logger.warning("HarborSolver: %s", error)
+                except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)